modules/library/main/src/main/java/org/geotools/geometry/jts/GeometryClipper.java

```diff
@@ -76,6 +79,49 @@ public GeometryClipper(Envelope bounds) {
     }
     
     /**
+     * This will try to handle failures when clipping - i.e. because of invalid input geometries (often 
+     * caused by simplification).
+     * 
+     * This attempts to do a normal clip().  If it fails, it will try to do more to ensure the clip 
+     * works properly.  
+     * 
+     * The first attempt to correct the geometry is to put the points on a precision grid and redo
+     * the clip. 
+     * If this fails, the original geometry is returned.
+     * 
+     * see {@link #clip(Geometry, boolean) clip}
+     * 
+     * @param g
+     * @param ensureValid
+     * @param scale 0=double precision Precision Model
+     * @return
+     */
+    public Geometry clipFailResistant(Geometry g, boolean ensureValid, double scale) {
```

Comment:
Maybe just leave it there and call it "tryClip" or whatever makes it evident that a clip is attempted, but might not succeed, and that will not make it throw an exception.
Or call it robustClip, and make it throw an exception in case of failure, that will be caught by the StreamingRenderer and just move on using the original geometry.